aspnetcore/mvc/advanced/custom-model-binding.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Custom Model Binding | Microsoft Docs
+author: ardalis
+description: Customizing model binding in ASP.NET Core MVC.
+keywords: ASP.NET Core, model binding, custom model binder
+ms.author: riande
+manager: wpickett
+ms.date: 4/10/2017
+ms.topic: article
+ms.assetid: ebd98159-a028-4a94-b06c-43981c79c6be
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: mvc/advanced/custom-model-binding
+---
+# Custom Model Binding
+
+By [Steve Smith](http://ardalis.com)
+
+Model binding allows controller actions to work directly with model types, passed in as method arguments, rather than HTTP requests. Repetitive mapping between incoming request data and application models is consolidated into model binders. Developers can extend the built-in model binding functionality by implementing custom model binders.
+
+[View or download sample from GitHub](https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/advanced/custom-model-binding/)
+
+## The problem
+
+The default model binding supports most of the common data types in the framework. The built-in model binders should support most of developers' needs. However, the default model binders expect to bind text-based input from the request directly to model types. You might need to transform the input prior to binding it. For example, when you have a key that can be used to look up model data. You can use a custom model binder to fetch data based on the key and pass the result as the action's argument.
+
+Model binding uses specific definitions for the types it operates on. A *simple type* is a type that should be converted from a single string in the input. A *complex type* is a type that should be converted from multiple input values. The framework determines the difference based on the existence of a `TypeConverter`. It's recommended to create a type converter if you have a simple `string` -> `SomeType` mapping that doesn't require external resources.
+
+## The solution
+
+Before creating your own custom model binder, it's worth reviewing how existing model binders are implemented. For example, the [ByteArrayModelBinder](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.modelbinding.binders.bytearraymodelbinder) can be used to convert base64-encoded strings into byte arrays, for storage as files or database BLOB fields.
+
+### Working with the ByteArrayModelBinder
+
+Base64-encoded strings can be used to represent binary data. For example, the following image can be encoded as a string.
+
+![dotnet bot](custom-model-binding/images/bot.png "dotnet bot")
+
+A small portion of the encoded string is shown in the following image:
```

Comment:
My initial intent with providing the base64 string somewhere in the article is so that if they run the sample, they have a string on hand that they can copy/paste into Postman and see the result (saved as a file).